docs/src/main/asciidoc/hibernate-panache.adoc

```diff
@@ -0,0 +1,1192 @@
+and guarantee that you did not make any typo in the entity names, their fields, the HQL/SQL syntax, or the name of
+their parameters.
+
+=== Using the repository
+
+Using the repository can be done by simply injecting it where you intend to use it:
+
+```java
+import jakarta.inject.Inject;
+import jakarta.transaction.Transactional;
+
+import java.time.LocalDate;
+import java.util.List;
+
+public class Code {
+
+    @Inject
+    Cat.Repo repo;
+
+    @Transactional
+    public void method() {
+        Cat cat = new Cat();
+        cat.name = "Lucky";
+        cat.birth = LocalDate.of(2015, 01, 12);
+        cat.breed = Cat.Breed.CUTE;
+
+        // Persist the cat
+        cat.persist();
+
+        // Make a change, no need to update it
+        cat.name = "Luckynou";
+
+        // Find our cat
+        cat = repo.findByName("Luckynou");
+
+        // Find cute cats
+        List<Cat> cuteCats = repo.findCute();
+
+        // Delete our cat
+        cat.delete();
+
+        // Delete queries
+        repo.deleteByName("Lucky");
+        repo.deleteHairless();
+    }
+}
+```
+
+Now, as an added benefit, you can also access the repository using a handy shortcut generated static method on the
+https://docs.jboss.org/hibernate/orm/7.1/introduction/html_single/Hibernate_Introduction.html#static-metamodel[generated Metamodel]
+of your entity, which is super useful for discovery of operations (just type `Cat_.repo().` and see all your methods)
+and to avoid making a detour outside of your method to add an injected field, such as in this startup code that
+will delete all your cats (don't do this in production!!):
+
+```java
+import io.quarkus.runtime.Startup;
+import jakarta.transaction.Transactional;
+
+public class OnStart {
+    @Startup
+    @Transactional
+    public void startupMethod() {
+        Cat_.repo().deleteAll();
+    }
+}
+```
+
+All the repositories you define as nested interfaces of your entity will be available under their generated metamodel
+class under accessor methods of the same name as the repository. This is strictly equivalent to injecting the repository
+type, and in fact uses CDI under the hood to look up the repository.
+
+== The `PanacheRepository` super type
+
+Just like in previous versions of Hibernate ORM and Hibernate Reactive with Panache, the `PanacheRepository` type comes
+packed with most of the operations you need to work on your entity, such as, out of the box:
+
+```java
+import io.quarkus.hibernate.panache.blocking.PanacheBlockingQuery;
+import jakarta.inject.Inject;
+import jakarta.transaction.Transactional;
+import org.hibernate.Session;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+
+public class Code {
+
+    @Inject
+    Cat.Repo repo;
+
+    @Transactional
+    public void repositoryOperations(Cat cat) {
+        // entity operations
+        repo.persist(cat);
+        repo.delete(cat);
```

Comment:
Oh, so you can do without the active record pattern, then?

Reply:
You can, yes. That's always been possible with repositories, though I've never seen the point.